### ISSUE TITLE
Getting currently rendered item index on viewport

### ISSUE THREAD

**Issue by elzdave** [oleggrishechkin/react-viewport-list#21]
Hi, thanks for creating this wonderful package! This package suits my needs perfectly, but it lacks one feature that makes it perfect: an event emitter to tell which index is being rendered on viewport. It would be nice if you could add this feature 😃

**Comment by oleggrishechkin** [comment]
Hello, thanks for your feedback.
I need some clarification.
Do you want to know what indexes was rendered or what indexes exactly in viewport now? It's different, because some items can be rendered outside of viewport (check `overscan` prop).

**Comment by elzdave** [comment]
Oops, forget to mention that. I need to know what index(es) rendered in the viewport. Is it possible?

**Comment by oleggrishechkin** [comment]
Yes, I guess it's possible. Can you describe your use case? Why you need it?

**Comment by elzdave** [comment]
I make a PDF viewer based on `react-pdf`, with the help of your package to render the pages efficiently. Why I need the capability to tell which index(es) are visible in the viewport? Because I need to show the current page number in navigation bar.

Previously, I've used react-window to accomplish this (getting in viewport page number), but it didn't render the PDF pages as good as yours because of its fixed styling and complex functions.

**Comment by oleggrishechkin** [comment]
@elzdave v5.3.0 with `onViewportIndexesChange` prop was published. You can try it now.

This callback will run when exactly rendered in viewport items were changed

`type OnViewportIndexesChange = (viewportIndexes: [number, number]) => void`

**Comment by elzdave** [comment]
Alright, will try and tell you about it soon. Thank you!

**Comment by elzdave** [comment]
Hey, just integrated your solution and it **works** perfectly!
![gambar](https://user-images.githubusercontent.com/36229438/185977689-743c134d-99d8-4e9e-ac02-63b17070ca7a.png)

Really thank you for your help and support! 🥳🥳

**Comment by elzdave** [comment]
Oh I realized one thing, When I actually in page 1, the viewport index returned 0 and 1 (page 1 and 2), but the viewport only shows index 0 (page 1), while index 2 (page 2) is outside the viewport, but rendered by "overscan". Is it normal? Because another page will render just fine. eg: rendering only page 2 will return index 1 1.

I managed to turn the overscan to 0, but the page scrolling sometimes flicker.

**Comment by oleggrishechkin** [comment]
I think it's a bug. And maybe I found where is a mistake.

What about flicker? Scroll jumps? Or item loop with hide/show?

**Comment by elzdave** [comment]
Oh okay, I'll wait for the patch 👍🏽

Flicker only found when overscan it set to 0, which is normal. On default overscan, there's no flicker, at least on my case. No scroll jumps found, and haven't tested with hide/show item loop.

**Comment by oleggrishechkin** [comment]
Fixed, try it now please, it may help. (Previously, start and end items will be ignored)

**Comment by elzdave** [comment]
@oleggrishechkin I've tested your new patch, and I confirm that the bug has been fixed, including one bug I haven't tell you about rendering array with single item. All good! Below is the result (showing currently visible PDF page(s) visible in viewport):

![halamansingle](https://user-images.githubusercontent.com/36229438/186115989-5cc31d9f-a013-4efd-8226-1b4371c1e058.png)
Rendering array with single item. Previously, the page says 0 (index -1). Fixed
![halaman6](https://user-images.githubusercontent.com/36229438/186116000-658491fc-cbd2-4ce5-be1f-e6ec6702c7e3.png)
Viewport shows one item.
![halaman56](https://user-images.githubusercontent.com/36229438/186116003-ca1518d7-5b7e-487e-8255-02d9ef39a674.png)
Viewport shows two items.
![halaman57](https://user-images.githubusercontent.com/36229438/186116010-7cb68d09-86c2-4585-8e41-9e2f42af3169.png)
Viewport shows three items.

No flicker, no glitch, just great! Thanks for the patch! 😃
